enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+
+- Proactively reach out to customers with "problem clusters"
+- Troubleshoot customer issues
+- Understand how individual applications are behaving at a holistic level
+
+### Goals
+
+- Provide OLM managed operators with a first-class way of creating `ServiceMonitors` and `PrometheusRules` objects to interact with the Prometheus Operator
+- Automatically integrating Red Hat approved Operators with the OpenShift Monitoring Prometheus Operator, allowing them to work with the Telemeter team to whitelist their metrics.
+
+### Non-Goals
+
+- Provide generic metrics about OLM managed operators
+- Whitelist OLM managed operator metrics so they are picked up by Telemeter.
+
+## Proposal
+
+### Generic Prometheus Operator Support
+
+#### OLM support for new Prometheus objects in bundles
+
+The bundle object currently has the following format:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC files. This enhancement proposes expanding this feature to support `ServiceMonitors` and `PrometheusRules` objects:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+With this change, OLM will need to be updated to manage the `ServiceMonitor` and `PrometheusRule` objects through the lifecycle of an operator.
+
+#### ServiceMonitor and PrometheusRule Support
+
+As OLM gains support for these Prometheus objects, there is a chance that installed operators may overload the Prometheus instance. It is important that a Cluster Admin is alerted that an operator will report metrics prior to deploying the operator. As such, we propose adding a `monitoringSupported` boolean field to the `ClusterServiceVersion` object. OLM will not install either of the Prometheus object unless the `monitoringSupported` field is set to true. Additionally, Cluster Admins will need to manually approve the operator's `InstallPlans` in the following scenarios:
+
+- A fresh install for this operator is being created where this field is set to `true`
+- An existing operator had this field set to `false` and is being upgraded to a new version that sets this field to `true`
+
+> Note: The `InstallPlan` will include a helpful message for Cluster Admins if either of the situations above are encountered.
+
+On upstream clusters, there is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the RBAC privileges to view the  `ServiceMonitor` and `PrometheusRule` object created alongsde the operator. As such, the cluster admin will need to update the correct Prometheus Operator's OperatorGroup to include the correct namespace.
```

Comment:
my understanding is the prometheus OLM operator is being deprecated

Reply:
I spoke with @sur and confirmed that this is true.

Based on the [user-workload-monitoring enhancement goals](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/user-workload-monitoring.md#goals) the integration points should still work.

Reply:
@awgreene What do you mean by "provided that this is available upstream".

Reply:
I am unsure what the definition of deprecating the prometheus OLM managed operator implies. Will vanilla Kuberenetes users with OLM installed still be able to deploy a Prometheus operator?

Reply:
We just deprecated support for OLM chart from our team. e.g. we are just not actively maintaining the OLM chart. In openshift we should **not** be able to install it via OLM, starting with 4.3 we have solutions for user monitoring.

As for vanilla Kuberenetes users, we are fine if there is Prometheus operator avaliable for install, we just cannot support the chart, same way as we do not support the Helm chart for Prometheus operator.

As I mentioned somewhere in this PR, Prometheus operator is not an upstream project, so all the custom resources OLM creates, that are Prometheus operator specific, there should be an option for users to disable it. e.g. `ServiceMonitor`. But as this is an openshift PR we assume we have those custom resources, but for upstream solution I would just provide for users a way to disable it. TBH k8s will just complain about not having that CRD registerted so not a big problem eitherway really.